diff --git a/CMakeLists.txt b/CMakeLists.txt
index dd9ed51b68d..ba485a6bc3a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,6 +6,10 @@
 # all out-of-source builds will use the CMakeCache.txt file there and even
 # build the libs and objects in it.
 
+if (WIN32)
+	set(CMAKE_SYSTEM_VERSION 10.0 (latest installed version))
+endif ()
+
 if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
   if(NOT DEFINED WITH_IN_SOURCE_BUILD)
     message(FATAL_ERROR
diff --git a/build_files/windows/check_libraries.cmd b/build_files/windows/check_libraries.cmd
index e8c04fb3258..6c0283e838e 100644
--- a/build_files/windows/check_libraries.cmd
+++ b/build_files/windows/check_libraries.cmd
@@ -35,6 +35,24 @@ if NOT EXIST %BUILD_VS_LIBDIR% (
 				echo.
 				exit /b 1
 			)
+
+:RETRYBCOMPAT
+			"%SVN%" checkout https://github.com/nalexandru/BlenderCompat/trunk/bin %BUILD_VS_LIBDIR%/bcompat7
+			if errorlevel 1 (
+				set /p LibRetry= "Error during download, retry? y/n"
+				if /I "!LibRetry!"=="Y" (
+					cd %BUILD_VS_LIBDIR%
+					"%SVN%" cleanup
+					cd %BLENDER_DIR%
+					goto RETRYBCOMPAT
+				)
+				echo.
+				echo Error: Download of external libraries failed.
+				echo This is needed for building, please manually run 'svn cleanup' and 'svn update' in
+				echo %BUILD_VS_LIBDIR% , until this is resolved you CANNOT make a successful blender build
+				echo.
+				exit /b 1
+			)
 		)
 	)
 ) else (
@@ -46,6 +64,8 @@ if NOT EXIST %BUILD_VS_LIBDIR% (
 			"%SVN%" update %BUILD_VS_LIBDIR%
 		)
 	)
+
+	"%SVN%" update %BUILD_VS_LIBDIR%/bcompat7
 )
 
 if NOT EXIST %BUILD_VS_LIBDIR% (
diff --git a/source/blender/gpu/opengl/gl_backend.cc b/source/blender/gpu/opengl/gl_backend.cc
index 98998e2e902..6ca98af96fb 100644
--- a/source/blender/gpu/opengl/gl_backend.cc
+++ b/source/blender/gpu/opengl/gl_backend.cc
@@ -295,8 +295,23 @@ static void detect_workarounds()
    * polaris platform. Keeping legacy platforms around just in case.
    */
   if (GPU_type_matches(GPU_DEVICE_ATI, GPU_OS_ANY, GPU_DRIVER_OFFICIAL)) {
-    const Vector<std::string> matches = {
-        "RX550/550", "(TM) 520", "(TM) 530", "(TM) 535", "R5", "R7", "R9"};
+    const Vector<std::string> matches = {"RX 460",
+                                         "RX 470",
+                                         "RX 480",
+                                         "RX 490",
+                                         "RX 560",
+                                         "RX 560X",
+                                         "RX 570",
+                                         "RX 580",
+                                         "RX 580X",
+                                         "RX 590",
+                                         "RX550/550",
+                                         "(TM) 520",
+                                         "(TM) 530",
+                                         "(TM) 535",
+                                         "R5",
+                                         "R7",
+                                         "R9"};
 
     if (match_renderer(renderer, matches)) {
       GCaps.use_hq_normals_workaround = true;
diff --git a/source/creator/CMakeLists.txt b/source/creator/CMakeLists.txt
index 7457358698d..1b06b28f538 100644
--- a/source/creator/CMakeLists.txt
+++ b/source/creator/CMakeLists.txt
@@ -32,6 +32,9 @@ if(WITH_TBB)
   list(INSERT LIB 0 bf_blenkernel)
 endif()
 
+link_directories(${LIBDIR}/bcompat7/lib)
+list(INSERT LIB 0 bcompat7)
+
 if(WIN32)
   blender_include_dirs(../../intern/utfconv)
 endif()
@@ -1011,6 +1014,12 @@ elseif(WIN32)
       DESTINATION ${TARGETDIR_VER}/python/lib/site-packages
     )
   endif()
+
+  install(
+    FILES ${LIBDIR}/bcompat7/bin/bcompat7.dll
+    DESTINATION "."
+    CONFIGURATIONS Release;RelWithDebInfo;MinSizeRel;Debug
+  )
 elseif(APPLE)
   if(NOT WITH_PYTHON_MODULE)
     # Uppercase name for app bundle
